app/events/domains/provider_domains_changed_event.rb

```diff
@@ -1,9 +1,13 @@
 class Domains::ProviderDomainsChangedEvent < BaseEventStoreEvent
-  def self.create(provider)
+  def self.create(provider, parent_event = nil)
     new(
+      parent_event_id: parent_event&.event_id,
```

Comment:
I dunno if it is on purpose (I guess not) but when there is no `parent_event`, this event will never be saved either... for this:
https://github.com/3scale/porta/blob/43383d8fe110b8ce25954a1748eb12b589966ace/app/services/notification_delivery_service.rb#L71-L73

Reply:
But that is for notifications only. This is not a notification and the test proves the event was created.

Reply:
> But that is for notifications only. This is not a notification.

Right, sorry, I see that you are right that it doesn't affect to this kind of events 😄 Although interestingly it seems to affect to `AccountDeletedEvent` although it is not a notification 🤔 but that last detail is an off-topic 😄 

> the test proves the event was created
The current test is not testing that it is persisted when this data is `nil` 😄 You are doing this (this is an example but all tests are like this):
```ruby
def test_provider_domain_changed_event
    provider = FactoryBot.create(:simple_provider)
    proxy = FactoryBot.create(:simple_proxy)
    proxy.update_attributes!(endpoint: "http://#{provider.admin_domain}")
    event = Domains::ProviderDomainsChangedEvent.create(provider)

    proxy_domains_changed = EventStore::Repository.adapter.where(event_type: 'Domains::ProxyDomainsChangedEvent')

    assert_difference proxy_domains_changed.method(:count) do
      ProcessDomainEventsWorker.new.perform(event)
    end
  end
```
But `event = Domains::ProviderDomainsChangedEvent.create(provider)` is not persisting, it is just being build (you can persist it with `Rails.application.config.event_store.publish_event(event)` and check its persitence for example with `assert EventStore::Event.where(event_id: event.event_id).last`), and `ProcessDomainEventsWorker.new.perform(event)` is testing the persistence but this is not the scenario in which `parent_event` is `nil`, because here parent event is the event you have just build 😄

Reply:
You are right 👍 Added those tests into the relevant event tests.